### PR TITLE
[118] reduce summary totals correctly

### PR DESCRIPTION
**Pull request by caravinci**
Some improvements made recently where displaying totals incorrectly in the UI (as described in #118), this solves those.

### CODE REVIEW

File: billing-ui/src/pages/Report/Report.js [hunk 1]
```diff
@@ -344,7 +344,7 @@ const Report = ({ history }) => {
                         {filters.bucketSize === 'YEARLY' && (
                             <AntdSelect
                                 defaultValue={moment(filters.fromDate).format('YYYY')}
-                                onChange={handleDateFromFilterChange}
+                                onChange={value => handleDateFromFilterChange(moment(value, 'YYYY'))}
```
Comment (caravinci) [inline billing-ui/src/pages/Report/Report.js:347]: Ensure the formatting for years

File: billing-ui/src/pages/Report/aggregateEntries.js [hunk 2]
```diff
@@ -34,8 +34,8 @@ const FIELDS = [
 ];
 
 const addNumbers = (acc, value, key) => {
-    const newNum = Number(formatNumber(value[key]));
-    const prevSum = Number(formatNumber(acc[key]));
```
Comment (caravinci) [inline billing-ui/src/pages/Report/aggregateEntries.js:38]: this generated NaNs in some edge cases when the user mistyped the dates

File: billing-ui/src/pages/Report/aggregateEntries.js [hunk 3]
```diff
@@ -44,12 +44,14 @@ const addNumbers = (acc, value, key) => {
...
         .map(([key, items]) => items.reduce((prevItems, entry) => ({
-            ...prevItems,
             ...entry,
             ...FIELDS.reduce((prevFields, fieldName) => ({
                 ...prevFields,
                 ...addNumbers(prevFields, entry, fieldName),
-            }), { key }),
+            }), {
+                key,
+                ...prevItems,
```
Comment (caravinci) [inline billing-ui/src/pages/Report/aggregateEntries.js:53]: I was merging the objects prematurely, and they were getting overwritten instead of summed up.

**Review by caravinci — COMMENTED** [review]
Some comments to help review

**Review by samrichca — APPROVED** [review]

**Pull request merged**